planner/core/logical_plan_builder.go

```diff
@@ -283,6 +283,12 @@ func (p *LogicalJoin) extractOnCondition(conditions []expression.Expression, der
 					continue
 				}
 			}
+
+			if binop.FuncName.L == ast.EQ && expression.IsEQCondFromIn(expr) {
+				// should I check null flag?
```

Comment:
I'm not sure about this. cc @eurekaka

Reply:
We don't need to check Null flag here now, since we have checked it already when marking `InOperand`. Can we move this branch to the front of the for loop and remove the `InOperand` check in line https://github.com/pingcap/tidb/pull/12075/files#diff-b9e75a99170ea35a67074d5d3d3305adR280?